pkg/aws/client.go

```diff
@@ -431,6 +436,54 @@ func (c *awsClient) GetRegion() string {
 	return aws.StringValue(c.awsSession.Config.Region)
 }
 
+func (c *awsClient) IsPublicSubnet(subnets []*ec2.Subnet) (map[string]bool, error) {
```

Comment:
Maybe we can go for something like `getPublicSubnetMap` or `generateSubnetToPublicMap`? as the function doesn't check if a single subnet is public but generates a mapping.